ocs_ci/deployment/azure.py

```diff
@@ -1,13 +1,80 @@
+# -*- coding: utf8 -*-
+"""
+This module contains platform specific methods and classes for deployment
+on Azure platform.
+"""
+
 import logging
 
+from ocs_ci.deployment.cloud import CloudDeploymentBase
+from ocs_ci.deployment.cloud import CloudIPIOCPDeployment
+from ocs_ci.utility.azure_utils import AZURE as AzureUtil
+
+
 logger = logging.getLogger(__name__)
 
 
-class AZUREIPI:
+__all__ = ['AZUREIPI']
+
+
+class AZUREBase(CloudDeploymentBase):
+    """
+    Azure deployment base class, with code common to both IPI and UPI.
+
+    Having this base class separate from AZUREIPI even when we have implemented
+    IPI only makes adding UPI class later easier, moreover code structure is
+    comparable with other platforms.
+    """
+
+    # default storage class for StorageCluster CRD on Azure platform
+    DEFAULT_STORAGECLASS = "managed-premium"
```

Comment:
shouldn't this be taken out to constants.py and called AZURE_DEFAULT_STORAGECLASS?

Reply:
I actually removed definitions of default storage class for each platform from constants module, so that the deployment details are stored in a single place: in deployment class of each platform.

Values of default storage class on each platform are used for deployment only anyway, and it also allows to simplify the process of adding new platform. To define a storage class for a platform prior this change, one had to update code in  places like `deployment.py` and `constants.py`. With this change, one just needs to make sure that deployment class for new platform has `DEFAULT_STORAGECLASS` attribute defined.

Reply:
OK. I understand, these are being used by the deployment class only